src/models/DataSource.ts

```diff
@@ -0,0 +1,217 @@
+/**
+ * LINKURIOUS CONFIDENTIAL
+ * Copyright Linkurious SAS 2012 - 2019
+ *
+ * - Created on 2019-09-02.
+ */
+import { GenericObject, ICaptionsConfig } from './Configuration';
+
+export interface IDataSourceParams {
+  sourceKey?: string;
+}
+
+export interface IDataSource {
+  name: string;
+  key: string;
+  configIndex: number;
+}
+
+export interface IDataSourceFeatures {
+  immutableNodeCategories: boolean;
+  canCountBeforeIndexation: boolean;
+  canIndexEdges: boolean;
+  searchHitsCount: boolean;
+  alternativeIds: GenericObject<string>;
+  maxNodeCategories?: number;
+  externalIndex: boolean;
+  canCount: boolean;
+  dialects: string[];
+}
+
+export interface ISpecialProperties {
```

Comment:
What is this ?

Reply:
let's ask @francescoinfante, `sources.settings.specialProperties`   -->     Properties with special access rights for this source